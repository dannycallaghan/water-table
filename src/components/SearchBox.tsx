import { memo, useState } from 'react';

const SearchBox = memo(function SearchBox(props: {
  handleSearch: (value: string) => void;
  term: string;
}) {
  const { handleSearch, term } = props;
  const [value, setValue] = useState<string>(term);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
    handleSearch(e.target.value.toLowerCase());
  };

  return (
    <>
      <div className="flex w-full max-w-56 flex-col">
        <label
          htmlFor="search-box"
          className="invisible text-[0.7rem] text-gray-400"
        >
          Search identifiers
        </label>
        <input
          id="search-box"
          placeholder="Search identifiers"
          className="h-8 w-full rounded border border-solid border-slate-300 p-1 text-[0.7rem] text-slate-600 focus:border-slate-300 focus:outline-none focus-visible:border-slate-300 active:border-slate-300"
          type="text"
          value={value}
          onChange={handleChange}
        />
      </div>
    </>
  );
});

export default SearchBox;
