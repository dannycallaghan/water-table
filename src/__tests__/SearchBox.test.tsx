import { expect, test, describe } from 'vitest';
import { render } from '@testing-library/react';
import SearchBox from '../components/SearchBox.tsx';

describe('SearchBox', () => {
  test('renders correctly', () => {
    const handleSearch = () => {};
    const { asFragment } = render(
      <SearchBox term={'Hello'} handleSearch={handleSearch} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
