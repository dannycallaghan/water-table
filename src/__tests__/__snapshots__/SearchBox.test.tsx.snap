// Vitest Snapshot v1

exports[`SearchBox > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex w-full max-w-56 flex-col"
  >
    <label
      class="invisible text-[0.7rem] text-gray-400"
      for="search-box"
    >
      Search identifiers
    </label>
    <input
      class="h-8 w-full rounded border border-solid border-slate-300 p-1 text-[0.7rem] text-slate-600 focus:border-slate-300 focus:outline-none focus-visible:border-slate-300 active:border-slate-300"
      defaultvalue="Hello"
      id="search-box"
      placeholder="Search identifiers"
      type="text"
    />
  </div>
</DocumentFragment>
`;
